internal/incident/incident.go

```diff
@@ -595,6 +643,117 @@ func (i *Incident) processAcknowledgementEvent(ctx context.Context, tx *sqlx.Tx,
 	return nil
 }
 
+// RetriggerEscalations tries to re-evaluate the escalations and notify contacts.
+func (i *Incident) RetriggerEscalations() {
+	i.Lock()
+	defer i.Unlock()
+
+	i.runtimeConfig.RLock()
+	defer i.runtimeConfig.RUnlock()
+
+	defer func() { i.scheduler.completed = true }()
+
+	ctx := context.TODO()
+	tx, err := i.db.BeginTxx(ctx, nil)
+	if err != nil {
+		i.logger.Errorw(
+			"failed to start time based incident escalations transaction", zap.String("object", i.ObjectDisplayName()),
+			zap.String("incident", i.String()), zap.Error(err),
+		)
+		return
+	}
+	defer func() { _ = tx.Rollback() }()
+
+	if err := i.ReloadRecipients(ctx); err != nil {
+		// The error is already logged within `i.ReloadRecipients()`
+		return
+	}
+
+	if !i.RecoveredAt.IsZero() {
+		// Incident is recovered in the meantime.
+		return
+	}
```

Comment:
This check should also be fine before performing any database query, the incident is locked so it shouldn't change.